### PR TITLE
[GLIMMER] Fix issues with actions with the `(mut` helper.

### LINKED ISSUE #13959: [Glimmer 2] (action (mut nonDefinedValue)) fails
Error:
> Action passed is null or undefined in (action)

Reproduction code:

```hbs
<input value={{nonDefinedProp}} oninput={{action (mut nonDefinedProp) value="target.value"}}>
```
_(note: the `value="target.value" is irrelevant`)_

Twiddle so you don't have to type the code above: 

https://ember-twiddle.com/56f80b7916bde51fb7d3ec47c7b39946?fileTreeShown=false&openFiles=templates.application.hbs%2C

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes #13959.

### CODE REVIEW

**Comment by zackthehuman** [comment]
LGTM. :+1:

File: packages/ember-glimmer/lib/modifiers/action.js [hunk 1]
```diff
@@ -194,7 +209,13 @@ export default class ActionModifierManager {
...
     let { positional } = args;
 
-    modifier.actionName = positional.at(1).value();
+    let actionNameRef = positional.at(1);
+
+    if (actionNameRef[INVOKE]) {
+      modifier.actionName = actionNameRef;
+    } else {
+      modifier.actionName = actionNameRef.value();
+    }
```
Comment (chancancode) [inline packages/ember-glimmer/lib/modifiers/action.js]: I don't fully understand this code yet, but the reference from `positional.at(1)` is a stable object (i.e. calling `position.at(1)` from here will give a `===` value from the same call in `install`). Like I said I need to keep reading to know what this is doing, but given the reference object itself cannot change, it's likely that you don't kneed to do any work here at all if `actionNameRef[INVOKE]` is truthy.

File: packages/ember-glimmer/lib/modifiers/action.js [hunk 2]
```diff
@@ -194,7 +209,13 @@ export default class ActionModifierManager {
...
     modifier.eventName = modifier.getEventName();
 
     // Not sure if this is needed? If we mutate the actionState is that good enough?
```
Comment (chancancode) [inline packages/ember-glimmer/lib/modifiers/action.js]: Would also like to know the answer to this

File: packages/ember-glimmer/tests/integration/helpers/element-action-test.js [hunk 3]
```diff
@@ -1489,4 +1489,47 @@ moduleFor('Helpers test: element action', class extends RenderingTest {
...
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(component.get('member'), 'foo', 'value is updated');
+
+    this.runTask(() => {
+      component.set('member', undefined);
+    });
```
Comment (chancancode) [inline packages/ember-glimmer/tests/integration/helpers/element-action-test.js]: Is there supposed to be another assertion after here?

File: packages/ember-glimmer/lib/modifiers/action.js (hunk 1 above)
Comment (rwjblue) [inline packages/ember-glimmer/lib/modifiers/action.js]: Thank you, that makes sense.  Will update.

File: packages/ember-glimmer/tests/integration/helpers/element-action-test.js (hunk 3 above)
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/helpers/element-action-test.js]: No, the failure scenario to start with was just having `{{action (mut someUndefinedValue)}}`. Setting back to `undefined` with a rerender is the "reset" step.

File: packages/ember-glimmer/lib/modifiers/action.js (hunk 2 above)
Comment (rwjblue) [inline packages/ember-glimmer/lib/modifiers/action.js]: I do not see the point of the lines below.  Seems a but odd, I think we can delete them.

File: packages/ember-glimmer/tests/integration/helpers/element-action-test.js (hunk 3 above)
Comment (chancancode) [inline packages/ember-glimmer/tests/integration/helpers/element-action-test.js]: I see. It seemed a bit odd to have that followed by nothing. Would it make sense to rewrite the tests as something like:

`+      template: '<button {{action (mut label) "foo"}}>{{if label label "Click me"}}</button>'`

So we could assert the value by checking the rendered values?

**Comment by chancancode** [comment]
:+1:

File: packages/ember-glimmer/lib/modifiers/action.js (hunk 2 above)
Comment (zackthehuman) [inline packages/ember-glimmer/lib/modifiers/action.js]: What if you bind the event name to a variable? How will the old handler be cleaned up unless you un/re register?

File: packages/ember-glimmer/tests/integration/helpers/element-action-test.js (hunk 3 above)
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/helpers/element-action-test.js]: Ya, good point.  Updated.

File: packages/ember-glimmer/lib/modifiers/action.js (hunk 2 above)
Comment (rwjblue) [inline packages/ember-glimmer/lib/modifiers/action.js]: I'm not sure what there is to clean up?

Comment (chancancode) [inline packages/ember-glimmer/lib/modifiers/action.js]: Merged the fix, but we should still figure out the answer to this question

**Pull request merged**